src/MeetingsSDKAdapter.js

```diff
@@ -0,0 +1,215 @@
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false,
+    };
+
+    sdkMeeting
+      .getMediaStreams(mediaSettings)
+      .then(([localStream, localShare]) => {
+        sdkMeeting
+          .addMedia({
+            localShare,
+            localStream,
+            mediaSettings,
+          })
+          .then()
+          .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(`Unable to add local media to meeting "${meetingID}"`, error);
+          });
+      })
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error('Unable to get local media streams', error);
+      });
+  }
+
+  /**
+   * Update the meeting object with media attached based on a given event type.
+   *
+   * @param {string} ID     ID of the meeting to fetch
+   * @param {Object} media  new media stream to attach to the meeting object based on a given event type
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  attachMedia(ID, {type, stream}) {
+    const meeting = this.meetings[ID];
+
+    switch (type) {
+      case MEDIA_TYPE_LOCAL:
+        this.meetings[ID] = {
+          ...meeting,
+          localAudio: stream.getAudioTracks()[0],
+          localVideo: stream.getVideoTracks()[0],
+        };
+        break;
+      case MEDIA_TYPE_REMOTE_AUDIO:
+        this.meetings[ID] = {...meeting, remoteAudio: stream};
+        break;
+      case MEDIA_TYPE_REMOTE_VIDEO:
+        this.meetings[ID] = {...meeting, remoteVideo: stream};
+        break;
+      default:
+        this.meetings[ID] = {...meeting};
+        break;
+    }
+  }
+
+  /**
+   * Creates meeting and returns an observable to the new meeting data.
+   *
+   * @param {string} target destination to start the meeting at
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsSDKAdapter
+   */
+  createMeeting(target) {
+    return from(this.datasource.meetings.create(target)).pipe(
+      map(
+        ({id, destination, sipuri}) => {
+          this.meetings[id] = {
+            ID: id,
+            title: destination || sipuri,
+            localVideo: null,
+            localAudio: null,
+            localShare: null,
+            remoteStream: null,
+            remoteShare: null,
+          };
+
+          return this.meetings[id];
+        },
+        (error) => {
+          // eslint-disable-next-line no-console
+          console.error(`Unable to create a meeting "${target}"`, error);
+        }
+      )
+    );
+  }
+
+  /**
+   * Returns a SDK meeting object retrieved from the collection.
+   *
+   * @param {string} ID ID of the meeting to fetch.
+   * @returns {Object} The SDK meeting object from the meetings collection.
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  fetchMeeting(ID) {
+    return this.datasource.meetings.getMeetingByType('id', ID);
+  }
+
+  /**
+   * Joins and returns an observable to the status of the connection.
```

Comment:
This should not return.
```suggestion
   * Attempts to join the meeting of the given meeting ID.
   * If the meeting is successfully joined, a ready event is dispatched.
```